utils/python/CIME/env_module.py

```diff
@@ -119,37 +120,65 @@ def _load_module_modules(self, modules_to_load):
             exec(py_module_code)
 
     def _load_soft_modules(self, modules_to_load):
-        logging.info("Loading soft modules")
         sh_init_cmd = self._machine.get_module_system_init_path("sh")
         sh_mod_cmd = self._machine.get_module_system_cmd_path("sh")
-
-        cmd = "source %s && source $SOFTENV_ALIASES && source $SOFTENV_LOAD" % (sh_init_cmd)
-
+        
+        # Some machines can set the environment
+        # variables using a script (such as /etc/profile.d/00softenv.sh
+        # on mira or /etc/profile.d/a_softenv.sh on blues)
+        # which load the new environment variables using softenv-load.
+
+        # Other machines need to run soft-dec.sh and evaluate the output,
+        # which may or may not have unresolved variables such as
+        # PATH=/soft/com/packages/intel/16/initial/bin:${PATH}
+
+        cmd = "source %s" % sh_init_cmd
+
+        if os.environ.has_key("SOFTENV_ALIASES"):
+            cmd += " && source $SOFTENV_ALIASES"
+        if os.environ.has_key("SOFTENV_LOAD"):
+            cmd += " && source $SOFTENV_LOAD"
+            
         for action,argument in modules_to_load:
             cmd += " && %s %s %s" % (sh_mod_cmd, action, argument)
 
         cmd += " && env"
-        logging.debug("Command=%s" % cmd)
         output=run_cmd(cmd)
 
+        ###################################################
+        # Parse the output to set the os.environ dictionary
+        ###################################################
         newenv = {}
+        dolater = []
         for line in output.splitlines():
-            m=re.match(r'^(\S+)=(\S+)$',line)
+            if line.find('$')>0:
```

Comment:
case.get_resolved_value() passes the string along to
GenericXML.get_resolved_value() which
can resolve strings using the current environment variables, but here I
wanted to resolve using
a separate dictionary that isn't fully defined until all the output has
been parsed.




On Wed, Jun 1, 2016 at 10:23 AM, jedwards4b <notifications@github.com>
wrote:

> In utils/python/CIME/env_module.py
> <https://github.com/ESMCI/cime/pull/203#discussion_r65382733>:
>
> >          for line in output.splitlines():
> > -            m=re.match(r'^(\S+)=(\S+)$',line)
> > +            if line.find('$')>0:
>
> Isn't this functionality already provided by case.get_resolved_value?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ESMCI/cime/pull/203/files/9a425ee1557fbc8f01669a8ed9d698016ded83bd#r65382733>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEs_uqCFfyz9Yi2DHSR1i3hMZ2eAPfmEks5qHaQAgaJpZM4IrEZG>
> .
>